### PR TITLE
fix update_mask in patch variable route

### LINKED ISSUE #29702: Updating Variables Description via PATCH in Airflow API is Clearing the existing description field of the variable and unable to update description field
### Apache Airflow version

2.5.1

### What happened

When i made these patch requests to update description of the variable via axios

### 1) Trying to modify new value and description
```javascript

let payload ={ key : "example_variable", value : "new_value", description: "new_Description" }
axios.patch("https://localhost:8080/api/v1/variables/example_variable" , payload ,
{
    auth : {
        username : "username",
        password : "password"
    },
    headers: {
        "Content-Type" : "application/json",
     }
});
```

following response received and In the airflow , Existing Variable's ```Description``` is cleared and set to ```None```

```html 
response body : {
    "description" : "new_Description",
    "key": "example_variable",
    "value" : "new_value"
}
```

### 2) Trying to update Description with update_mask
```javascript

let payload ={ key : "example_variable", value : "value", description: "new_Description" }
axios.patch("https://localhost:8080/api/v1/variables/example_variable?update_mask=description" , payload ,
{
    auth : {
        username : "username",
        password : "password"
    },
    headers: {
        "Content-Type" : "application/json",
     }
});
```

following response received

```html 
response body : {
    "detail" : null,
    "status": 400,
    "detail" : "No field to update",
    "type" : "https://airflow.apache.org/docs/apache-airflow/2.5.0/stable-rest-api-ref.html#section/Errors/BadRequest"
}
```

### What you think should happen instead

The filed "description" is ignored both while setting the Variable (L113) and ```update_mask``` (L107-111).

https://github.com/apache/airflow/blob/1768872a0085ba423d0a34fe6cc4e1e109f3adeb/airflow/api_connexion/endpoints/variable_endpoint.py#L97-L115

Also in Variable setter its set to ```None``` if input doesn't contain description field
https://github.com/apache/airflow/blob/1768872a0085ba423d0a34fe6cc4e1e109f3adeb/airflow/models/variable.py#L156-L165

### API call
"https://localhost:8080/api/v1/variables/example_variable?update_mask=description"

### payload
{ key : "example_variable", value : "value", description: "new_Description" }

### headers
"Content-Type" : "application/json"

OR

### API call
"https://localhost:8080/api/v1/variables/example_variable

### payload
{ key : "example_variable", value : "new_value", description: "new_Description" }

### headers
"Content-Type" : "application/json"

### Operating System

Ubuntu 22.04.1 LTS

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else

Possible Solution to update Description field can be like

https://github.com/apache/airflow/blob/1768872a0085ba423d0a34fe6cc4e1e109f3adeb/airflow/api_connexion/endpoints/connection_endpoint.py#L134-L145

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #29702 

---
`patch_variable` route does not allow updating the variable description. This PR updates the method and uses the same pattern used in the other routes to dynamically update all the fields which are not `non_update_fields`.

### CODE REVIEW

File: airflow/api_connexion/endpoints/variable_endpoint.py [hunk 1]
```diff
@@ -103,15 +109,22 @@ def patch_variable(*, variable_key: str, update_mask: UpdateMask = None) -> Resp
...
-    Variable.set(data["key"], data["val"])
-    return variable_schema.dump(data)
+        update_mask = [i.strip() for i in update_mask]
+        data_ = {}
+        for field in update_mask:
+            if field in data and field not in non_update_fields:
+                data_[field] = data[field]
+            else:
+                raise BadRequest(detail=f"'{field}' is unknown or cannot be updated.")
+        data = data_
```
Comment (uranusjr) [inline airflow/api_connexion/endpoints/variable_endpoint.py]:
```suggestion
        data_ = {}
        for field in update_mask:
            field = field.strip()
            if field in data and field not in non_update_fields:
                data_[field] = data[field]
            else:
                raise BadRequest(detail=f"'{field}' is unknown or cannot be updated.")
        data = data_
```

File: airflow/api_connexion/endpoints/variable_endpoint.py [hunk 2]
```diff
@@ -103,15 +109,22 @@ def patch_variable(*, variable_key: str, update_mask: UpdateMask = None) -> Resp
...
+    for key in data:
+        setattr(variable, key, data[key])
```
Comment (uranusjr) [inline airflow/api_connexion/endpoints/variable_endpoint.py]:
```suggestion
    for key, val in data.items():
        setattr(variable, key, val)
```

**Review by uranusjr — COMMENTED** [review]
Minor code improvements

File: airflow/api_connexion/endpoints/variable_endpoint.py [hunk 3]
```diff
@@ -103,15 +109,22 @@ def patch_variable(*, variable_key: str, update_mask: UpdateMask = None) -> Resp
...
-    Variable.set(data["key"], data["val"])
-    return variable_schema.dump(data)
+        data_ = {}
+        for field in update_mask:
+            field = field.strip()
+            if field in data and field not in non_update_fields:
+                data_[field] = data[field]
+            else:
+                raise BadRequest(detail=f"'{field}' is unknown or cannot be updated.")
+        data = data_
```
Comment (dimberman) [inline airflow/api_connexion/endpoints/variable_endpoint.py]: This seems like a lot of nesting. Can you break this out into a separate function?

**Review by dimberman — DISMISSED** [review]

**Comment by hussein-awala** [comment]
@dimberman Can you please review my last commit?

File: airflow/api_connexion/endpoints/variable_endpoint.py [hunk 4]
```diff
@@ -103,15 +110,15 @@ def patch_variable(*, variable_key: str, update_mask: UpdateMask = None) -> Resp
...
-        if "value" not in update_mask:
-            raise BadRequest("No field to update")
-
-    Variable.set(data["key"], data["val"])
-    return variable_schema.dump(data)
+        data = extract_update_mask_data(update_mask, non_update_fields, data)
+    for key, val in data.items():
+        setattr(variable, key, val)
+    session.add(variable)
+    session.commit()
```
Comment (uranusjr) [inline airflow/api_connexion/endpoints/variable_endpoint.py]: Don’t need this, the session is comitted when the function returns.

**Comment by potiuk** [comment]
@dimberman ? Is your "request changes" still valid ? I think it has been addressed but I would not want to dismiss the review without you confirming.

**Comment by potiuk** [comment]
Since you have not responded @dimberrman, I will simply dismiss it. 2 weeks and no response despite pinging seems to be enough. We can always have a follow-up if the "request for changes" is still not resolved.

**Review by potiuk — APPROVED** [review]

**Pull request merged**